pkg/collector/metrics_collector.go

```diff
@@ -0,0 +1,28 @@
+package collector
+
+import (
+	"go.uber.org/atomic"
+)
+
+// MetricsCollector is a simple collector that can be used to accumulate a metric
+// we primarily use it to collect the total bytes read from a reader across a request
+type MetricsCollector struct {
```

Comment:
Have some thoughts on this.  (a) Wondering if we can simplify this to something like `type MetricsCollector atomic.Uint64`. And it still provides the Add and Load methods.  (b) or else question about if the field should be pointer?  Because the line `*atomic.NewUint64(0)` doesn't match other uses of atomics.

Reply:
updated.